nmoscommon/utils.py

```diff
@@ -54,3 +78,82 @@ def getLocalIP():
                 pass
     # Could not find an interface
     return None
+
+
+def downgrade_api_version(
+    obj, rtype, target_ver, downgrade_ver=None, downgrade_map=DOWNGRADE_MAP, remove_keys=["max_api_version"]
+):
+
+    # Sort version list in ascending order
+    version_list = sorted(downgrade_map.keys())
+    # Add 'v1.0' as first element in version list
+    version_list.insert(0, 'v1.0')
+    # Set the max version as last element in list
+    max_ver = version_list[-1]
+
+    # Get current API version, or default to max API version
+    current_api_version = obj.get("@_apiversion", max_ver)
```

Comment:
The previous default may have been to use 'v1.0' if not present. I wonder if given that we're writing this fresh now it might be better to assume @_apiversion is always present and throw an exception if not? This would avoid the need for this code and a little more further down.

Reply:
This is the only thing yet to resolve as it sounds like it requires a bit of tinkering with the individual API implementations (Node & Query) considering `max_api_version` only exists in the Node API and `@_apiversion` is only in the query API